trio/_core/__init__.py

```diff
@@ -44,7 +40,22 @@ def _public(fn):
 
 from ._local import RunVar
 
-if hasattr(_run, "wait_readable"):
+# Linux and OSX imports
+try:
+    from ._run import (wait_readable, wait_writable, notify_fd_close)
+
     wait_socket_readable = wait_readable
     wait_socket_writable = wait_writable
     notify_socket_close = notify_fd_close
+except ImportError:
+    pass
+
+# Windows imports
+try:
+    from ._run import (
+        wait_socket_readable, wait_socket_writable, notify_socket_close,
+        monitor_completion_key, current_iocp, register_with_iocp,
+        wait_overlapped, write_overlapped, readinto_overlapped
+    )
+except ImportError:
+    pass
```

Comment:
As noted in gitter, the reason the tests are failing on macOS is that you forgot to export the kqueue symbols in this file :-)